lib/tasks/govuk_publishing_components_tasks.rake

```diff
@@ -1,4 +1,28 @@
-# desc "Explaining what the task does"
-# task :govuk_publishing_components do
-#   # Task goes here
-# end
+desc 'validate component documentation'
+task :validate_component_documentation do
+  components_missing_docs = []
+
+  Dir["app/views/#{GovukPublishingComponents::Config.component_directory_name}/**/*.html.erb"].each do |view_file|
+    component_doc_match = false
+
+    Dir["app/views/#{GovukPublishingComponents::Config.component_directory_name}/docs/**/*.yml"].each do |doc_file|
```

Comment:
This could be optimised by caching the results of `Dir["app/views/#{GovukPublishingComponents::Config.component_directory_name}/docs/**/*.yml"]` to avoid looping through all the doc files for each partial.

Reply:
You could probably do something like:

```ruby
partials = Dir["app/views/#{GovukPublishingComponents::Config.component_directory_name}/**/*.html.erb"]

partials.each do |partial|
  expected_documentation = view_file.gsub('.html.erb', '.yml') #or something
  component_missing_docs << view_file unless File.exist?(expected_documentation_file)
end
```